internal/manager.go

```diff
@@ -669,12 +671,83 @@ func (m *Manager) ParseLegacyFile(output func(sdkname, version string)) error {
 					}
 				}
 			}
-		}
+			return nil
+		})
 
 	}
 	return nil
 }
 
+func (m *Manager) FindToolVersion() (*util.SortedMap[string, string], error) {
+	path := m.PathMeta.WorkingDirectory
+	legacyFileRecord, err := m.loadLegacyFileRecord()
+	if err != nil {
+		return nil, err
+	}
+	versionMap := util.NewSortedMap[string, string]()
+	for {
+		logger.Debugf("Find tool version in %s \n", pterm.LightBlue(path))
+		file := filepath.Join(path, toolset.ToolVersionFilename)
+		if util.FileExists(file) {
+			logger.Debugf("Parsing tool version file: %s \n", file)
+			wtv, err := toolset.NewFileRecord(file)
+			if err != nil {
+				return nil, err
+			}
+			versionMap = wtv.SortedMap
+		}
+		if m.Config.LegacyVersionFile.Enable {
+			var wg sync.WaitGroup
+			for _, filename := range legacyFileRecord.Keys() {
+				wg.Add(1)
+				go func(fn string) {
+					defer wg.Done()
+					sdkname, _ := legacyFileRecord.Get(fn)
+					legacyFile := filepath.Join(m.PathMeta.WorkingDirectory, fn)
+					if util.FileExists(legacyFile) {
+						logger.Debugf("Parsing legacy file %s \n", legacyFile)
+						if sdk, err := m.LookupSdk(sdkname); err == nil {
+							// The .tool-version in the current directory has the highest priority,
+							// checking to see if the version information in the legacy file exists in the former,
+							//  and updating to the former record (Don’t fall into the file!) if it doesn't.
+							if version, err := sdk.ParseLegacyFile(legacyFile); err == nil && version != "" {
+								logger.Debugf("Found %s@%s in %s \n", sdkname, version, legacyFile)
+								versionMap.Set(sdkname, string(version))
+							}
+						}
+					}
+				}(filename)
+			}
+			wg.Wait()
+		}
+		if versionMap.Len() != 0 {
```

Comment:
I may not understand the code correctly. As soon as the first .toolsversion is found, the recursion is stopped. However, it could be that not all SDKs are specified in this file. In other words, I would expect a combination of all .toolsversions from all paths. The global file has the lowest priority and then ascending according to proximity to the current working directory.

Reply:
You reminded me that the global `.tool-versions` file should be combined.

> In other words, I would expect a combination of all .toolsversions from all paths

I don't agree to this point.

Reply:
> I don't agree to this point.

Expressed incorrectly. I mean all files that can be found up to the root. Say the following example

![image](https://github.com/version-fox/vfox/assets/13434739/f6f679b6-d914-448e-8669-ccd026484bdf)

If I enter folder childdir1, I will receive all SDK in `.tool-versions` file 1 and 3, but not 2.

Reply:
Sorry for the late reply. I have got your means, and I will continue to improve this PR soon later.